ui/lib/core/addon/components/form-field.js

```diff
@@ -81,6 +81,31 @@ export default class FormFieldComponent extends Component {
     this.toggleInputEnabled = !!modelValue;
   }
 
+  // ---------------------------------------------------------------
+  // IMPORTANT:
+  // this controls the top-level logic in the associated template
+  // to decide which type of form field to render (Vault or HDS)
+  // ---------------------------------------------------------------
+  //
+  get isHdsFormField() {
+    const { type, options } = this.args.attr;
+
+    // here we replicate the logic in the template, to make sure we don't change the order in which the "ifs" are evaluated
+    if (options?.possibleValues?.length > 0) {
```

Comment:
This whole conditional logic will be little by little populated, depending on the component that are migrated to HDS. If you want to see an example have a look at this similar (spike) PR: https://github.com/hashicorp/vault/blob/21a890a58eeb0f31235b0a9f476b120cf89f9d6f/ui/lib/core/addon/components/form-field.js#L90-L106